test/iht/forms/registration/DeceasedFormsTest.scala

```diff
@@ -208,93 +225,170 @@ class DeceasedFormsTest extends FormTestHelper with FakeIhtApp {
     }
 
     "give an error when the last name is too long" in {
       val data = completeAboutDeceased + ("lastName" -> "A value that's longer than the 40 characters allowed in this field")
       val expectedErrors = error("lastName", "error.lastName.giveUsingXCharsOrLess")
 
-      checkForError(aboutDeceasedForm()(messages), data, expectedErrors)
+      checkForError(deceasedForms.aboutDeceasedForm()(messages, createFakeRequest(true), hc, ec), data, expectedErrors)
     }
 
     "give an error when a blank marital status is supplied" in {
       val data = completeAboutDeceased + ("maritalStatus" -> "")
       val expectedErrors = error("maritalStatus", "error.invalid")
 
-      checkForError(aboutDeceasedForm()(messages), data, expectedErrors)
+      checkForError(deceasedForms.aboutDeceasedForm()(messages, createFakeRequest(true), hc, ec), data, expectedErrors)
     }
 
     "give an error when an invalid marital status is supplied" in {
       val data = completeAboutDeceased + ("maritalStatus" -> "INVALID")
       val expectedErrors = error("maritalStatus", "error.invalid")
 
-      checkForError(aboutDeceasedForm()(messages), data, expectedErrors)
+      checkForError(deceasedForms.aboutDeceasedForm()(messages, createFakeRequest(true), hc, ec), data, expectedErrors)
     }
 
     "give an error when no marital status is supplied" in {
       val data = completeAboutDeceased - "maritalStatus"
       val expectedErrors = error("maritalStatus", "error.deceasedMaritalStatus.select")
 
-      checkForError(aboutDeceasedForm()(messages), data, expectedErrors)
+      checkForError(deceasedForms.aboutDeceasedForm()(messages, createFakeRequest(true), hc, ec), data, expectedErrors)
     }
 
     "give multiple errors when several fields are invalid" in {
-      val data = completeAboutDeceased + ("firstName" -> "", "nino" -> "INVALID")
+      val data = completeAboutDeceased + ("firstName" -> "", "lastName" -> "")
       val expectedErrors = error("firstName", "error.firstName.give") ++
-        error("nino", "error.nino.giveUsing8Or9Characters")
+        error("lastName", "error.lastName.give")
+      val form = deceasedForms.aboutDeceasedForm()(messages, createFakeRequest(true), hc, ec)
 
-      checkForError(aboutDeceasedForm()(messages), data, expectedErrors)
+      checkForError(form, data, expectedErrors)
     }
 
     "give one date error when several date fields are invalid" in {
       val data = completeAboutDeceased + ("dateOfBirth.day" -> "32", "dateOfBirth.month" -> "13", "dateOfBirth.year" -> "12", "dateOfBirth.day" -> "99")
       val expectedErrors = error("dateOfBirth", "error.dateOfBirth.giveFull")
 
-      checkForError(aboutDeceasedForm()(messages), data, expectedErrors)
+      checkForError(deceasedForms.aboutDeceasedForm()(messages, createFakeRequest(true), hc, ec), data, expectedErrors)
+    }
+
+    "indicate validation error when nino for coexecutor validation fails" in {
```

Comment:
Change this comment to read "..for deceased validation fails...", i.e. not coexecutor.